### PR TITLE
Sdb unit test for sorting big lists

### PR DESCRIPTION
**Pull request by GovanifY**
For now this unit test triggers a crash, related to radare/radare2#7079

### CODE REVIEW

**Comment by radare** [comment]
```
test_sdb.c:54:16: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'const char *' [-Wint-conversion]
                sdb_set (db, i-1, i, 0);
                             ^~~
../../src/sdb.h:146:31: note: passing argument to parameter 'key' here
int sdb_set(Sdb*, const char *key, const char *data, ut32 cas);
                              ^
test_sdb.c:54:21: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'const char *' [-Wint-conversion]
                sdb_set (db, i-1, i, 0);
                                  ^
../../src/sdb.h:146:48: note: passing argument to parameter 'data' here
int sdb_set(Sdb*, const char *key, const char *data, ut32 cas);
                                               ^
2 warnings generated.
```

**Comment by radare** [comment]
thats why it segfaults :P

**Comment by radare** [comment]
fixed your test case and its not crashing at all this is not a valid test for your issue

**Comment by radare** [comment]
also you are not checking if the resulting list is sorted or not

**Comment by GovanifY** [comment]
I thougt we didn't had in that case as long as it didn't crashed. Argh back to reproducing then, sorry >_>

**Comment by radare** [comment]
i am sorting 10 million items in this SdbList without any stack exhaustion or crash problem

**Comment by radare** [comment]
```
diff --git a/test/unit/test_sdb.c b/test/unit/test_sdb.c
index 95f29f6..4d76430 100644
--- a/test/unit/test_sdb.c
+++ b/test/unit/test_sdb.c
@@ -50,11 +50,12 @@ static int __cmp_asc(const void *a, const void *b) {
 bool test_sdb_list_big(void) {
        Sdb *db = sdb_new (NULL, NULL, false);
        int i;
-       for ( i=1; i < 100000; i++) {
-               sdb_set (db, i-1, i, 0);
+       for ( i=0; i < 10000; i++) {
+               sdb_num_set (db, sdb_fmt (0, "%d", i), i + 1, 0);
        }
        SdbList *list = sdb_foreach_list (db, true);
-       ls_sort(list, __cmp_asc);
+       ls_sort (list, __cmp_asc);
+// TODO: verify if its sorted
        sdb_free (db);
        mu_end;
 }
```

**Comment by GovanifY** [comment]
Patch applied, any idea on how to reproduce the issue then?

**Comment by GovanifY** [comment]
Also confirm it works here too even though that was to be expected: `test_sdb_list_big OK`

**Comment by GovanifY** [comment]
Got a segfault at 10M here, will try to reproduce at lower count and add here. Tooked a _lot_ of time tho

**Comment by radare** [comment]
10 million items. The hash is a table of lists, maybe some stats to see how many items are in each bucket may help to identify the issue

**Comment by radare** [comment]
But maybe we have reached a limit in mmap

**Comment by GovanifY** [comment]
Test is pretty slow in its current form but works™

**Comment by GovanifY** [comment]
There is a chance, for mmap, I'll check that later, thanks for pointing out

File: test/unit/test_sdb.c [hunk 1]
```diff
@@ -41,6 +41,25 @@ bool test_sdb_list_delete(void) {
...
+
+bool test_sdb_list_big(void) {
+	Sdb *db = sdb_new (NULL, NULL, false);
+	int i;
+	for ( i=0; i < 1000000; i++) {
+    	sdb_num_set (db, sdb_fmt (0, "%d", i), i + 1, 0);
+	}
+	SdbList *list = sdb_foreach_list (db, true);
+	ls_sort (list, __cmp_asc);
+	// TODO: verify if its sorted
```
Comment (alvarofe) [inline test/unit/test_sdb.c]: Here you are sorting twice since sdb_foreach_list with the true flag already does the same operation as above.

**Comment by alvarofe** [comment]
Working on this btw

**Comment by alvarofe** [comment]
Fixed in master. I lowered the amount to sort to `500000` just to save time when testing. 

Test again please and tell me.

**Pull request closed without merging**